### ISSUE TITLE
Retrospective for January 2022 Releases

### ISSUE THREAD

**Issue by adamfarley** [adoptium/adoptium#104]
**Summary**

A retrospective for all efforts surrounding the titular releases. 

All community members are welcome to contribute to the agenda via comments in the linked Google Document \(subsequently deleted, with contents copied [here](https://github.com/adoptium/adoptium/issues/104#issuecomment-1036326893)\).

This will be a virtual meeting after the release, with at least a week of notice in the #temurin-compliance Slack channel.

On the day of the meeting the agenda items in the Google Document will be iterated over, with a list of actions added at the end.

Invited: Everyone.

**Time, Date, and URL**

Time: 2:30pm GMT (9:30am Ontario)
Date: Friday the 11th of February

URL: https://meet.google.com/jvg-deeg-fwh

**Details**

Retrospective Owner Tasks (in order):

- [x] Post this issue's URL on the \#Release Slack channel around the start of the new release.
- [x] Copy actions from the previous retrospective into this issue, while ignoring actions that are ticked or have issue links (i.e. clearly-completed actions).
- [x] Wait until 90% of all builds have been released, with no clear signs of additional respins.
- [x] Announce the retrospective Slack call's date + time on \#Release at least one full week in advance, and send out meeting invites.
- [x] Host the slack call for the retrospective, including:
  - [x] Iterating over the actions from the previous retrospective issue, ticking off completed items.
  - [x] Iterate over the agenda, ensuring everything gets debated.
  - [x] Create a clear list of actions at the end of the retrospective, including volunteer names.
- [x] Create a new retrospective issue for the next release.
- [ ] Copy the \"Next Retrospective\" Google document into the relevant subfolder and rename it appropriately.
- [ ] Empty the Retrospective_Next Google document (except for the actions, which should be starred), and change the month at the top to the next release month.
- [x] Set yourself a calendar reminder so that you remember to commence step 1 (in the new issue) just before the next release.
- [x] Close this issue.

Add proposed agenda items to [this Google Document](https://docs.google.com/document/d/1JPIvjl__YJbMpSYyKND1s-rPVOlS_V1k3vzooTIDaNs/edit?usp=sharing).

**Comment by sxa** [comment]
Memo to self - RELEASING.md doc updates for:
- Ensuring mirrors are run and have the tags before initiating pipelines

**Comment by karianna** [comment]
[Triage.md](https://github.com/adoptium/aqa-tests/blob/master/doc/Triage.md) has some out of date instructions.  Especially around having to install a CL tool to parse TRSS output.

**Comment by karianna** [comment]
TRSS seems to have the ability to add labels to the builds in the dashboard.  It appears that is surfaced via the override publish name field when you fire off the pipeline.  Worth adding that to the instructions, e.g. jdk-11.0.14-special-martijn-build and have it default to SCM Ref

**Comment by karianna** [comment]
In TRSS I would like to be able to sort the possibleIssues table by Status.

**Comment by karianna** [comment]
When I'm in TAP Results in Jenkins I click on the failing test e.g. https://ci.adoptopenjdk.net/job/Test_openjdk11_hs_extended.openjdk_arm_linux/30/tapTestReport/ and it doesn't send me to any details :-(

**Comment by karianna** [comment]
In TAP Extended View in Jenkins I wanted to click through to the TAP results https://ci.adoptopenjdk.net/job/Test_openjdk11_hs_extended.openjdk_arm_linux/30/artifact/Test_openjdk11_hs_extended.openjdk_arm_linux_testList_2/24/aqa-tests/TKG/output_16425491068250/Test_openjdk11_hs_extended.openjdk_arm_linux_testList_2.tap/*view*/ but that URL was a 404

**Comment by karianna** [comment]
The tiny link icon in TRSS output that takes us to Jenkins is non obvious (but saved my butt when I found it!).  Can we make it super obvious please.

**Comment by karianna** [comment]
Some slight inconsistency with aqa-tests issues being left open with a known JBS and closed with a known JBS.  https://github.com/adoptium/aqa-tests/issues/2382 vs https://github.com/adoptium/aqa-tests/issues/2992

**Comment by karianna** [comment]
TRSS Report reports the parent build as unstable/failure but this isn't 'useful' per se during triage.  Looking at the individual test failure categories is:

Test_openjdk11_hs_extended.openjdk_x86-64_mac ❌ FAILURE ❌ - N/A - This is a parent build, actual failures are listed below

**Comment by karianna** [comment]
Could we have TRSS summary report the successes as well?  So we know that certain combinations did actually run and were successful.

**Comment by smlambert** [comment]
> Could we have TRSS summary report the successes as well? So we know that certain combinations did actually run and were successful.

We already have that one in plan (see https://github.com/adoptium/aqa-test-tools/issues/582).

**Comment by sxa** [comment]
Failures with the mirror jobs should probably post into Slack somewhere (Although the one hour schedule might make that quite chatty!)

**Comment by sxa** [comment]
Updates to release tool docs e.g. examples of:
- **/*x64_mac*.tar.gz,**/*x64_mac*sha256.txt,**/*x64_windows*.zip,**/*x64_windows*.sha256.txt,**/*x64_windows*.msi,**/*x64_mac*.pkg,**/*x64_windows*.json,**/*x64_mac*.json
- **/*x64_linux*.tar.gz,**/*x64_linux*.sha256.txt,**/*x64_linux*.json,**/*source*
And artefacts to skip of:
- **/*testimage*

**Comment by sxa** [comment]
Observation - likely not fatal but can be confusing to see lines like this interleaved in the output of the release jobs:
`version:17 type: -jdk arch:x64 os:linux variant:hotspot timestamp or version:tar.gz timestamp: extension:`

**Comment by sxa** [comment]
This will need some enhancement/automation but as a check that the correct set of artifacts have gone up I've been using this quick and dirty script as a very basic check of the count of artifacts in the release for each platform (Note that it dosn't skip platforms that aren't on all release so e.g. macos/aarch64 will always show as "Not complete" for JDK8)

Here is the output before and after shipping Linux/arm32 and Win32 for JDK11:
```
[sxa@sainz bin]$ ./releaseCheck.sh 11 jdk-11.0.14+9
Grabbing information from https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.14+9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  278k    0  278k    0     0   230k      0 --:--:--  0:00:01 --:--:--  230k
Linux on x64: OK!
Linux on aarch64: OK!
Linux on ppc64le: OK!
Linux on s390x: OK!
Linux on arm: Not complete:
Windows on x64: OK!
Windows on x86-32: Not complete:
MacOS on x64: OK!
MacOS on aarch64: Not complete:
Source images: OK!
[sxa@sainz bin]$ ./releaseCheck.sh 11 jdk-11.0.14+9
Grabbing information from https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.14+9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  289k    0  289k    0     0   283k      0 --:--:--  0:00:01 --:--:--  283k
Linux on x64: OK!
Linux on aarch64: OK!
Linux on ppc64le: OK!
Linux on s390x: OK!
Linux on arm: OK!
Windows on x64: OK!
Windows on x86-32: OK!
MacOS on x64: OK!
MacOS on aarch64: Not complete:
Source images: OK!
[sxa@sainz bin]$ 
```
It can take a third parameter (doesn't matter what it is) to show you the list of matching artifacts for anything listed as not complete in case just one is missing. Script is here: [releaseCheck.sh.txt](https://github.com/adoptium/adoptium/files/7960638/releaseCheck.sh.txt). There's a lot more than can be done with this, but it's a good "status check" during the release.

**Comment by sxa** [comment]
Find a way to adjust the processes so that we can ship Linux installers without having to have all the new versions for every architecture available. arm32 is going to hold us back for this release.

**Comment by sxa** [comment]
Does anyone use the links in the status documents for downloading the binaries or finding the links for re-runs of test jobs?

**Comment by smlambert** [comment]
re: https://github.com/adoptium/adoptium/issues/104#issuecomment-1026641327 - if you mean links in the Github issues that track status then 100% of the time that is where I pull links from

**Comment by sxa** [comment]
Need to sort the checksums `sha256.txt` where some of them seem to have a full path to the file and others don't (Correct option is NOT to have a path name in there). Causes problems for the Linux installer processes (and is also just wrong IMHO)

**Comment by sxa** [comment]
>> Does anyone use the links in the status documents for downloading the binaries or finding the links for re-runs of test jobs?

> re: [#104 (comment)](https://github.com/adoptium/adoptium/issues/104#issuecomment-1026641327) - if you mean links in the Github issues that track status then 100% of the time that is where I pull links from

Fair enough, although they were never added for 8 and 17 in https://github.com/adoptium/adoptium/issues/109 and no-one has objected ;-)

**Comment by sxa** [comment]
Need to make sure that the releasing guide is populated with up to date information. If part of the process is changed or has additions the work should not be considered complete until the releasing guide is in sync.

**Comment by smlambert** [comment]
Was raised somewhere in Slack that TRSS release report shows undefined for some child jobs of the build pipeline (this is due to the permissions it has to see various job status).  This issue is raised to broaden the read permissions of eclipse-temurin-bot: https://github.com/adoptium/ci-jenkins-pipelines/issues/236.

**Comment by smlambert** [comment]
re: https://github.com/adoptium/adoptium/issues/104#issuecomment-1016365317 - created https://github.com/adoptium/aqa-tests/issues/3327 to address (as good first issue for OpenForce hackathon in March).

**Comment by sxa** [comment]
Good use of time zone differences with @sxa and @smlambert triaging and re-running failed JCK automated tests (Seemed especially effective for 11.0.14.1)

**Comment by sxa** [comment]
Consider adding a dry run (i.e. no `PUBLISH` checkbox) of the installer pipeline onto the end of the weekly runs to validate it regularly. Also perhaps rename the option to `DRY_RUN` and reverse the logic to make it consistent with the publish pipeline.

**Comment by sxa** [comment]
Can we do anything with the linuxNew packaging job to make it more obvious when it is blocked on a machine not being available so remedial action can be taken? (Note: Currently looking at the BlueOcean view seems to be the best way, or possibly pipeline steps)

**Comment by adamfarley** [comment]
Should we continue to use the Google document?

**Comment by adamfarley** [comment]
Adamf: Actions from this retro (will tidy up later today).

Actions:

@@ Andrew
https://github.com/adoptium/adoptium/issues/104#issuecomment-1017354900

@@ Stewart
https://github.com/adoptium/adoptium/issues/104#issuecomment-1017747613
https://github.com/adoptium/adoptium/issues/104#issuecomment-1021303106

to chat w shelley: https://github.com/adoptium/adoptium/issues/104#issuecomment-1026641327

@@ Adam

Raise issue build repo
https://github.com/adoptium/adoptium/issues/104#issuecomment-1023656434

Add to download artifacts issue action thingy
https://github.com/adoptium/adoptium/issues/104#issuecomment-1024443867

create issue installer repo if doesn't exist already
https://github.com/adoptium/adoptium/issues/104#issuecomment-1026073992

build issue
https://github.com/adoptium/adoptium/issues/104#issuecomment-1031454239

raise issue to discuss in trss repo: https://github.com/adoptium/adoptium/issues/104#issuecomment-1032581533

pipeline issue
https://github.com/adoptium/adoptium/issues/104#issuecomment-1035336197

carry over to next retro
https://github.com/adoptium/adoptium/issues/104#issuecomment-1036065992

To remove Google doc.

**Comment by adamfarley** [comment]
Closing in favour of the [next retrospective](https://github.com/adoptium/adoptium/issues/128) for clarity. Unticked actions will be copied over into the new issue closer to the next retrospective \(there's a tickable item in the new issue to ensure this is not forgotten\).

**Comment by sxa** [comment]
- Improvements to how to use the release tool (Maybe have platform dropdowns?): https://github.com/adoptium/temurin-build/issues/2847
- Make it easier to determine if linuxNew installer job is stalled on a particular machine/platform: https://github.com/adoptium/installer/issues/453
